test/libsolidity/syntaxTests/specialFunctions/abidecode/abi_decode_member_acess.sol

```diff
@@ -0,0 +1,19 @@
+library L {
+    struct S { int a; }
+    enum State { idle, running, blocked }
+}
+
+contract D {
+    struct X { uint b; }
+    enum Color { red, green, blue }
+}
+
+contract C {
+    function f() pure public {
+        abi.decode("", (L.S));
```

Comment:
Huh, how come we don't get `Statement has no effect` warnings here?

Reply:
I was surprised by it too, but this behaviour was not introduced by the PR, so I didn't investigate it.

Reply:
Interesting. Looks like it's the index access that somehow bypasses it.